### PR TITLE
Migrate Provisioner tests to Go modules 

### PR DESCRIPTION
**Pull request by franpog859**
**Description**

Changes proposed in this pull request:

- migrate Provisioner tests to Go modules 
- bump Go version in Provisioner

### CODE REVIEW

**Review by rafalpotempa — APPROVED** [review]

File: components/provisioner/hack/gqlgen.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// +build ignore
+// +build tools
```
[the remark this answers is not in the document]
Comment (franpog859) [reply components/provisioner/hack/gqlgen.go:1]: Yeah! Thanks for your help here ^^

**Pull request merged**